### DISCUSSION TITLE
use Diataxis as the framework for mkdocs' docs

### DISCUSSION THREAD

**Discussion by MatthewSteen** [mkdocs/mkdocs#3557] [Ideas]
I find navigating the MkDocs' docs difficult and suggest reorganizing it using https://diataxis.fr/, which is a widely used opinionated framework for documentation.

**Comment by dmbr0** [comment]
I second this. I can't really find any documentation software that follows this.

**Reply by MatthewSteen** [reply]
> I second this. I can't really find any documentation software that follows this.

One documentation software that follows it is https://jupyterbook.org/. 

Many other non-documentation software uses it too. For example. https://docs.djangoproject.com/.

**Reply by dmbr0** [reply]
Thank you. I guess I mean I can't find an open source self hosted one.

**Comment by facelessuser** [comment]
This seems more a suggestion for a specific theme in how it styles the content in a page as this isn't something controlled by MkDocs itself.

**Reply by MatthewSteen** [reply]
To be clear, this is a suggestion for organizing MkDocs' own documentation at https://www.mkdocs.org/. It's not a suggestion for a theme or style.

**Reply by facelessuser** [reply]
I think it would require a change to the MkDocs theme to provide similar results. So, I guess the two are tied together.

**Reply by MatthewSteen** [reply]
Really? Again, to clarify further I'm not suggesting following the style or theme of the Diataxis website, I'm suggesting using its framework. 

So, I think it just requires the docs to be organized into the four Diataxis categories of the following, which could replace the top-level categories across the top of the MkDocs site.
- Explanation
- Guides
- Reference
- Tutorials

**Reply by Andre601** [reply]
I think the term "framework" is confusing everyone here and what you mean is "structure", as what you're suggesting is effectively restructuring the docs themself.

**Reply by MatthewSteen** [reply]
@Andre601 yes, that could be since it's not a "web framework".

**Reply by dmbr0** [reply]
I agree. I think we are suggesting a theme/interface that follows and puts the documentation into the suggested structure of diataxis.

**Reply by pawamoy** [reply]
> I think the term "framework" is confusing everyone

I'm kinda surprised that one can be confused, to be honest. The Diataxis docs are pretty clear, and framework sounds like a good choice of word for what it is :slightly_smiling_face: 

> I think we are suggesting a theme/interface that follows and puts the documentation into the suggested structure of diataxis.

Only the second part, yes. @MatthewSteen suggests that MkDocs follows the Diataxis framework for its documentation. The theme/interface can stay absolutely the same as the current one.

**Reply by MatthewSteen** [reply]
@pawamoy yeah, I think it's just renaming and reorganizing dirs and files in the docs dir, along with updating the config yml.

**Comment by pawamoy** [comment]
@MatthewSteen it would help if could point out what exactly you found difficult to navigate, for example if you were looking for specific info and had trouble finding it, or if you expected a different flow of pages.

@dmbr0 not sure I understand your comment. Did you mean "that *doesn't* follow" the Diataxis framework?
